### PR TITLE
[DOCFIX] Update upgrade docs

### PR DESCRIPTION
**Pull request by LuQQiu**

### CODE REVIEW

File: docs/en/operation/Upgrade.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+In normal cases, users can directly shutdown the current Alluxio processes, change
+the Alluxio binaries to newer version, configure Alluxio clusters similar to before, 
+and start Alluxio processes with the existing journal folder/address for upgrading.
+Alluxio is able to read the previous journal files and recover the Alluxio metadata.
+
+However, Alluxio has two versions that are not backward compatible on the journal management.
+The two versions are Alluxio 2.0.0 and Alluxio 2.4.0 which results in the journals across one of these versions
+are not compatible.
```
Comment (bf8086) [inline docs/en/operation/Upgrade.md:16]:
```suggestion
There are two cases where master journals are not backward compatible and additional steps are required to upgrade Alluxio cluster:

- Upgrading from Alluxio 1.x to Alluxio 2.x
- Upgrading from Alluxio 2.3.x and below to Alluxio 2.4.0 and above, and the embedded journal is used.
```

File: docs/en/operation/Upgrade.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+This document goes over how to upgrade Alluxio across those non-compatible versions.
+Even if Alluxio journal is compatible between the current version and the target version to upgrade to, 
+the upgrade method provided in the later sections is still preferred and less error-prone.
```
Comment (bf8086) [inline docs/en/operation/Upgrade.md:20]:
```suggestion
This document goes over how to upgrade Alluxio to a non-backward compatible version.
Even when upgrading to a backward compatible version, it is still recommended to follow the steps below to create a backup before upgrading.
```

**Review by bf8086 — CHANGES_REQUESTED** [review]
Thanks, I added some minor comments.

**Comment by LuQQiu** [comment]
@bf8086 PTAL, thanks

**Review by bf8086 — APPROVED** [review]
LGTM.

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**